src/main/java/io/zksync/signer/ZkSigner.java

```diff
@@ -42,21 +47,49 @@ private ZkSigner(ZksPrivateKey privateKey, ZksCrypto crypto) {
     }
 
     public static ZkSigner fromSeed(byte[] seed) {
+        try {
+            // Generate private key from seed
+            ZksPrivateKey privateKey = crypto.generatePrivateKey(seed);
 
-        // Load native library
-        ZksCrypto crypto = ZksCrypto.load();
+            return new ZkSigner(privateKey);
+        } catch (ZksSeedTooShortException e) {
+            throw new ZkSyncException(e);
+        }
+    }
 
+    public static ZkSigner fromRawPrivateKey(byte[] rawPrivateKey){
         try {
-
             // Generate private key from seed
-            ZksPrivateKey privateKey = crypto.generatePrivateKey(seed);
+            ZksPrivateKey privateKey = crypto.generatePrivateKey(rawPrivateKey);
+            privateKey.data = rawPrivateKey;
```

Comment:
Maybe there is a cleaner way to implement this? Calling the function that is used to generate the private key from seed and then assigning it's field a value directly looks a little bit weird.

Reply:
I'm agree but unfortunately ZksCrypto library has no constructor for building private key instance from raw bytes. I think I will add the constructor in the next release of ZksCrypto